### ISSUE TITLE
Unused ProcessUtil library

### ISSUE THREAD

**Issue by dangercrow** [Healix/Gw2Launcher#298]
Hi, I note the ProcessUtil project (https://github.com/Healix/Gw2Launcher/tree/master/ProcessUtil) appears to be unused -- is that expected?

Can it just be deleted from the repo?

The Gw2Launcher project appears to have most of the functionality that it needs in Gw2Launcher.Util.ProcessUtil, so I guess its leftover unused code?

**Comment by Healix** [comment]
Yes, it's not used. All of that was merged into Gw2Launcher practically from the beginning.

**Comment by dangercrow** [comment]
I've made an MR to remove this. I'd like to contribute more code to this repo if you're willing, @Healix.

In particular around modernising the .NET version and release process -- that way we can have more standardised build processes and pipelines in GitHub instead of you building and committing the EXE.

I know you have some code somewhere for the Beta, is that just locally?
Could you push it to a branch, perhaps?

**Comment by Healix** [comment]
Everything here is basically an old copy, always, as I have a master build that everything is created from. I use a tool to automatically fill in variable information and strip out anything I've marked, like notes and tools that have no purpose for anyone else. That stripped copy becomes the release build here. I left in ProcessUtil simply for the timestamp.

The beta version is not something I normally do. Outside of GW2 breaking something, the beta version isn't really a beta version, it's specifically to test something, like when someone finds a bug. The next release is basically ready, I've just been wasting time on experimenting with CEF.

As for trying to change anything, don't bother. I'd just have to manually do it all.

**Comment by dangercrow** [comment]
What's the reason for having this as a public repo & permissively licensed codebase if you don't intend to have collaborative improvements?

I can see some significant drawbacks to just randomly running an EXE that 'someone on the internet' claims is built from this source, as well as development woes with using a version of .NET that is 12 years old (and EOL'd 7 years ago).

If you're pained by doing things manually -- why not modernise? Automate -- or have someone (such as me) in your OSS community help?

**Comment by Healix** [comment]
Everything here _is_ automated from my perspective. This is effectively a read-only public copy of my private work. The license is because someone asked for one. When I said I'd have to manually do it, I meant any changes would only serve as a suggestion that I would do myself. Feel free to make your own copy from it, but I'm only doing this for the fun of it and if I let someone else work on it, I'd only lose interest and abandon it.

As for the version, I know people were relying on 4.5 under Linux. The highest I have is 4.7.2. It takes no effort to change it, so I'll switch it for the next release and see if anyone complains.

**Comment by dangercrow** [comment]
FYI I have a working port of the (non-beta) code on .NET 4.8 which is the latest .NET framework at https://github.com/dangercrow/Gw2Launcher
